docs/topics/api-guidelines-backward-compatibility.md

```diff
@@ -0,0 +1,318 @@
+
+```text
+0: invokestatic  #12 // Method Library.demo:()Ljava/lang/Number;
+```
+
+The JVM is trying to invoke a static method called demo which returns a `Number`.
+However, as this method no longer exists, you have broken binary compatibility.
+
+## Avoid using data classes in your API
+
+In regular development, the strength of data classes is the extra functions that are generated for you.
+In API design, this strength becomes a weakness.
+
+For example, let’s say you use the following data class in your API:
+
+```kotlin
+data class User(
+    val name: String,
+    val email: String
+)
+```
+
+Later, you might want to add a property called `active`:
+
+```kotlin
+data class User(
+    val name: String,
+    val email: String,
+    val active: Boolean = true
+)
+```
+
+This would break binary compatibility in two ways. Firstly, the generated constructor will have a different signature.
+Additionally, the signature of the generated `copy` method changes.
+
+The original signature (on Kotlin/JVM) would be:
+
+```text
+public final User copy(java.lang.String, java.lang.String)
+```
+
+After adding the `active` property, the signature becomes:
+
+```text
+public final User copy(java.lang.String, java.lang.String, boolean)
+```
+
+As with the constructor, this breaks binary compatibility.
+
+It’s possible to work around these issues by manually writing a secondary constructor and overriding the `copy` method.
+However, the effort involved negates the convenience of using a data class.
+
+Another issue with data classes is that changing the order of constructor arguments affects the generated `componentX` methods,
+which are used for destructuring. Even if it does not break binary compatibility, changing the order will definitely break behavioral compatibility.
+
+## Considerations for using the PublishedApi annotation
+
+Kotlin allows inline functions to be a part of your library’s API. Calls to these functions will be inlined into the
+client code written by your users. This can introduce compatibility issues, so these functions are not allowed to call non-public-API declarations.
+
+If you need to call an internal API of your library from an inlined public function, you can do so by annotating it with [`@PublishedApi`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-published-api/).
+This makes the internal declaration effectively public, as references to it will end up in compiled client code.
+Therefore, it must be treated the same as public declarations when making changes to it, as these changes might affect binary compatibility.
+
+## Evolve APIs pragmatically
+
+There are cases where you need to make breaking changes to your library’s API over time by removing or changing an existing declaration.
+In this section, we’ll discuss how to handle such cases pragmatically.
+
+When users upgrade to a newer version of your library, they should not end up with unresolved references to your library’s
```

Comment:
find+replace ’ to '

Reply:
ah nice catch! - replaced all occurences